### PR TITLE
Fix slide url + refacto

### PR DESCRIPTION
**Pull request by SharakPL**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Adds support for all types of link, including the one starting with `/`
| Type?         | bug fix
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes PrestaShop/PrestaShop#31721
| How to test?  | 1. In BO go to Modules > ps_imageslider<br>2. Change the slide's link<br>3. Check the link set on this slide in FO.<br><br>Links to test:<br>`http://prestashop.com/en/download`<br>`https://prestashop.com/en/download`<br>`//prestashop.com/en/download`<br>`prestashop.com/en/download`<br>`/content/4-about-us`<br>`content/4-about-us`<br>`index.php`<br>`search?controller=search&s=mug`<br>`#footer`<br><br>Expected result for shop domain `localhost:8001`:<br>`http://prestashop.com/en/download`<br>`https://prestashop.com/en/download`<br>`//prestashop.com/en/download`<br>`//prestashop.com/en/download`<br>`http://localhost:8001/content/4-about-us`<br>`http://localhost:8001/content/4-about-us`<br>`http://localhost:8001/index.php`<br>`http://localhost:8001/search?controller=search&s=mug`<br>`#footer`

### CODE REVIEW

File: ps_imageslider.php [hunk 1]
```diff
@@ -583,8 +590,9 @@ protected function updateUrl($link)
             return $link;
         }
 
-        if (substr($link, 0, 7) !== 'http://' && substr($link, 0, 8) !== 'https://') {
-            $link = 'http://' . $link;
+        $url = parse_url($link);
+        if (empty($url['host'])) {
+            $link = $this->context->link->getBaseLink() . ltrim($link, '/');
```
Comment (SharakPL) [inline ps_imageslider.php:595]: This is the fix. The rest is automatic refacto.

**Review by okom3pom — APPROVED** [review]

**Review by leemyongpakvn — DISMISSED** [review]

File: ps_imageslider.php (hunk 1 above)
Comment (leemyongpakvn) [reply ps_imageslider.php:595]: with $link = 'www.prestashop.com':
$url['host'] is empty while $url['path'] = 'www.prestashop.com'

Comment (SharakPL) [reply ps_imageslider.php:595]: I know and I have fix for that ready. Will add it soon...

**Comment by SharakPL** [comment]
Changed the function name, but it's used only in this file so it shouldn't be considered a BC break.

**Review by Hlavtox — APPROVED** [review]

**Review by MhiriFaten — COMMENTED** [review]
Hello @SharakPL ,

I have checked your PR : 

**PS8.0.1** :heavy_check_mark: 
The issue is well fixed.

**PS1.7.8.8** :x: 

[Untitled_ Mar 14 2023 11_35 AM.webm](https://user-images.githubusercontent.com/111756615/224974463-f286a2bd-ac88-4339-bf10-3951e4ee85c0.webm)

The image slide still with the old URL after editing and saving (Same problem with Title, Caption and description)

Could you checked it please?
Thank you.

**Comment by SharakPL** [comment]
@MhiriFaten you saved url for english version 

obraz

but checked the result on french

obraz

Please check again for correct version.

**Review by MhiriFaten — APPROVED** [review]
Yes sorry that was it !

Thank you! so it is QA approved :heavy_check_mark:

**Comment by SharakPL** [comment]
Thanks @MhiriFaten :)

**Comment by kpodemski** [comment]
thanks @SharakPL 🎉

**Pull request merged**